java/src/jmri/configurexml/BlockManagerXml.java

```diff
@@ -86,120 +84,104 @@ public Element store(Object o) {
             // (and this (now way overly verbose) comment).
             if (true) {
                 // write out first set of blocks without contents
-                while (iter.hasNext()) {
+                for (Block b : blkList) {
+                    if (b == null) {
+                        log.error("Block null during store1, skipped");
+                        break;
```

Comment:
What is the intent here? Should the break quit the "for" loop or continue to the next item in blkList? For me, it seems that the "break" statement should be a "continue" statement instead.

Reply:
It's not supposed to ever happen, as the system name list should never contain a null.  I recommend removing it, or (better) switching the iteration over to the NamedBeanSet and then removing it.

Reply:
Done, a NamedBeanSet can be empty but can't contain null elements via ``register()``